### PR TITLE
Refactored System info handlers

### PR DESCRIPTION
**Pull request by Ynng**
# Description

Part of #51 but doesn't close it yet.

Refactored system info handlers and converted them to axum style.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

*Note: make sure your files are formatted with rust-analyzer*

### CODE REVIEW

File: src/handlers/system.rs [hunk 1]
```diff
@@ -1,75 +1,112 @@
...
     let sys = System::new();
-    match sys.cpu_load_aggregate() {
-        Ok(load) => {
-            thread::sleep(Duration::from_secs(1));
-            return (Status::Ok, load.done().unwrap().user.to_string())
-        },
-        Err(_) => return (Status::BadRequest, "failed to get cpu info".to_string())
-    }
+    let cpu_aggregate = sys.cpu_load_aggregate().expect("Failed to get cpu load");
+    thread::sleep(Duration::from_secs(1));
```
Comment (CheatCod) [inline src/handlers/system.rs]: use thread::sleep blocks the async runtime, use one of the async sleep

File: src/handlers/system.rs [hunk 2]
```diff
@@ -1,75 +1,112 @@
...
+    let cpu_load = cpu_aggregate.done().expect("Failed to get cpu load");
+
+    Json(CPUInfo {
+        cpu_vendor: "TODO".to_string(),
```
Comment (CheatCod) [inline src/handlers/system.rs]: extra info is sent on handler that is periodic pulled

File: src/handlers/system.rs [hunk 3]
```diff
@@ -1,75 +1,112 @@
...
+        cpu_speed: cpu_speed().unwrap(),
```
Comment (CheatCod) [inline src/handlers/system.rs]: use expect() instead of unwrap() unless you are 100% sure this won't fail

**Review by CheatCod — CHANGES_REQUESTED** [review]

File: src/handlers/system.rs (hunk 2 above)
Comment (Ynng) [reply src/handlers/system.rs]: removed

Comment (Ynng) [reply src/handlers/system.rs]: just remember to implement it later

**Review by CheatCod — APPROVED** [review]

**Pull request merged**